Engine/source/gui/controls/guiTextCtrl.cpp

```diff
@@ -188,7 +188,7 @@ void GuiTextCtrl::setText(const char *txt)
    if( !mProfile )
       return;
    
-   if (txt)
+   if (txt && txt != mText)
```

Comment:
It's a pointer comparison, not a string compare. strncpy was being called with identical src and dest pointers, which is a no-no according to the C standard (although it happens to work with most implementations). I believe this happens when GuiTextEditCtrl calls setText(mText). I wasn't sure what the purpose of that was, but I assume it was being done for a reason, so I added the pointer check in order to avoid the violation and error report by asan.